src/c++/perf_analyzer/client_backend/triton_c_api/triton_c_api_backend.h

```diff
@@ -0,0 +1,183 @@
+/// triton inference service. This uses the local C++ library
+///
+class TritonCApiClientBackend : public ClientBackend {
+ public:
+  /// Create a triton client backend which can be used to interact with the
+  /// server.
+  /// \param triton_server_path Tritonserver library that contains
+  /// lib/libtritonserver.so.
+  /// \param model_repository_path The model repository.
+  /// \param memory_type Type of memory used in Triton Server.
+  /// \param verbose Enables the verbose mode of TritonServer.
+  /// \param client_backend Returns a new TritonCApiClientBackend object.
+  /// \return Error object indicating success
+  /// or failure.
+  static Error Create(
+      const std::string& triton_server_path,
+      const std::string& model_repository_path, const std::string& memory_type,
+      const bool verbose, std::unique_ptr<ClientBackend>* client_backend);
+
+  ~TritonCApiClientBackend() { TritonLoader::Delete(); }
+
+  /// See ClientBackend::ServerExtensions()
+  Error ServerExtensions(std::set<std::string>* server_extensions) override;
+
+  /// See ClientBackend::ModelMetadata()
+  Error ModelMetadata(
+      rapidjson::Document* model_metadata, const std::string& model_name,
+      const std::string& model_version) override;
+
+  /// See ClientBackend::ModelConfig()
+  Error ModelConfig(
+      rapidjson::Document* model_config, const std::string& model_name,
+      const std::string& model_version) override;
+
+  /// See ClientBackend::Infer()
+  Error Infer(
+      InferResult** result, const InferOptions& options,
+      const std::vector<InferInput*>& inputs,
+      const std::vector<const InferRequestedOutput*>& outputs) override;
+
+  /// See ClientBackend::ClientInferStat()
+  Error ClientInferStat(InferStat* infer_stat) override;
+
+  /// See ClientBackend::ModelInferenceStatistics()
+  Error ModelInferenceStatistics(
+      std::map<ModelIdentifier, ModelStatistics>* model_stats,
+      const std::string& model_name = "",
+      const std::string& model_version = "") override;
+
+ private:
+  TritonCApiClientBackend() : ClientBackend(BackendKind::TRITON_C_API) {}
+  void ParseInferInputToTriton(
+      const std::vector<InferInput*>& inputs,
+      std::vector<nic::InferInput*>* triton_inputs);
+  void ParseInferRequestedOutputToTriton(
+      const std::vector<const InferRequestedOutput*>& outputs,
+      std::vector<const nic::InferRequestedOutput*>* triton_outputs);
+  void ParseInferOptionsToTriton(
+      const InferOptions& options, nic::InferOptions* triton_options);
+  void ParseStatistics(
+      const rapidjson::Document& infer_stat,
+      std::map<ModelIdentifier, ModelStatistics>* model_stats);
+  void ParseInferStat(
+      const nic::InferStat& triton_infer_stat, InferStat* infer_stat);
+};
+
+//==============================================================
+/// TritonCApiInferInput is a wrapper around InferInput object of
+/// triton client library.
+///
+class TritonCApiInferInput : public InferInput {
```

Comment:
I think I am okay with the usage of `nic::InferInput` in TritonCApiInferInput . It is indeed a lite wrapper for client InferInput object, which is currently tuned for HTTP and gRPC protocol. That said I don't see any major perf drawback in it being used as a holder for input properties. On the other hand, using Trtion C API with our client library structure may help us discovering possible tune-ups.